helpers/telemetryHelper.js

```diff
@@ -0,0 +1,140 @@
+const Telemetry = require('sb_telemetry_util')
+const telemetry = new Telemetry()
+const _ = require('lodash')
+const fs = require('fs')
+const path = require('path')
+appId = 'discussionMiddleware'
+
+const telemtryEventConfig = JSON.parse(fs.readFileSync(path.join(__dirname, './telemetryEventConfig.json')))
+
+
+module.exports = {
+  logTelemetryErrorEvent: function (req, res, data, proxyResData, proxyRes, resCode) {
+    const context = {
+      env: 'discussion-middleware'
+    }
+    let telemetryObj = {};
+    if (proxyRes.statusCode === 404) {
+      if (data !== 'Not Found' && (typeof data) !== 'string') {
+        telemetryObj = JSON.parse(proxyResData.toString('utf8'));
+      } else {
+        telemetryObj = resCode;
+      }
+    } else {
+      if (resCode.params) {
+        telemetryObj = resCode;
+      } else {
+        telemetryObj = JSON.parse(proxyResData.toString('utf8'));
+      }
+    }
+    const option = telemetry.getTelemetryAPIError(telemetryObj, proxyRes, context);
+    if (option) { this.logApiErrorEventV2(req, telemetryObj, option) }
+  },
+
+  logApiErrorEventV2: function (req, data, option) {
+    let object = data.obj || {}
+    let channel =  _.get(req.headers,'x-channel-id')
+    const context = {
```

Comment:
This code is duplicated.. Why can't we write the getContext method which will retrurn the context Object for any event?

Reply:
ok